src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -65,12 +76,10 @@ void ClusterResourceScheduler::Init(
       local_node_resources,
       get_used_object_store_memory,
       get_pull_manager_at_capacity,
-      [&](const NodeResources &local_resource_update) {
+      [this](const NodeResources &local_resource_update) {
         cluster_resource_manager_->AddOrUpdateNode(local_node_id_, local_resource_update);
       });
-  if (!local_node_id_.IsNil()) {
-    cluster_resource_manager_->AddOrUpdateNode(local_node_id_, local_node_resources);
-  }
+  cluster_resource_manager_->AddOrUpdateNode(local_node_id_, local_node_resources);
```

Comment:
Why remove `if (!local_node_id_.IsNil())`?

Reply:
Gcs server would use this constructor, which uses null node_id:

`  // This constructor is only used by gcs scheduler. So the
  // `local_node_id` has to be nil and `is_node_available_fn`
  // has to exclude the local node.
  ClusterResourceScheduler(scheduling::NodeID local_node_id,
                           std::function<bool(scheduling::NodeID)> is_node_available_fn);`

Reply:
In gcs, we would create a cluster_resource_scheduler with Nil node id. I've slightly fixed this and added more explanations there.